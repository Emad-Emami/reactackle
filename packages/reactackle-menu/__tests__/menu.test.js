import React from 'react';
import renderer from 'react-test-renderer';
import { IconCross } from 'reactackle-icons';

import {
    Menu,
    MenuGroup,
    MenuList,
    MenuItem,
} from '../src';

jest.mock('react-dom');

const data = [
  {
    data: [
      {
        text: 'Structure',
      },
      {
        text: 'Design',
        submenuGroup: {
          data: [
            {
              data: [
                { text: 'Submenu 1-1' },
                { text: 'Submenu 1-2' },
              ],
            },
            {
              data: [
                { text: 'Submenu 2-1' },
                { text: 'Submenu 2-2' },
              ],
            },
          ],
        },
      },
    ],
  },
];


describe('<Menu/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <Menu />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop inline set true', () => {
    const tree = renderer.create(
      <Menu inline />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop colorScheme set dark', () => {
    const tree = renderer.create(
      <Menu colorScheme='dark' />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if contains children', () => {
    const tree = renderer.create(
      <Menu>
        <MenuGroup>
          <MenuList>
            <MenuItem text="test" />
          </MenuList>
        </MenuGroup>
      </Menu>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<MenuList/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <MenuList />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop bordered set true', () => {
    const tree = renderer.create(
      <MenuList bordered />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop bordered set false', () => {
    const tree = renderer.create(
      <MenuList bordered={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop title set', () => {
    const tree = renderer.create(
      <MenuList title="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop className set', () => {
    const tree = renderer.create(
      <MenuList className="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop prefix set', () => {
    const tree = renderer.create(
      <MenuList prefix="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('<MenuItem/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <MenuItem />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop iconLeft set', () => {
    const tree = renderer.create(
      <MenuItem iconLeft={<IconCross />} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop iconRight set', () => {
    const tree = renderer.create(
      <MenuItem iconRight={<IconCross />} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop textRight set', () => {
    const tree = renderer.create(
      <MenuItem textRight="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop image set', () => {
    const tree = renderer.create(
      <MenuItem image="star" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop linkHref set', () => {
    const tree = renderer.create(
      <MenuItem linkHref="#" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop text set', () => {
    const tree = renderer.create(
      <MenuItem text="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop prefix set', () => {
    const tree = renderer.create(
      <MenuItem prefix="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop className set', () => {
    const tree = renderer.create(
      <MenuItem className="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop textSecondary set', () => {
    const tree = renderer.create(
      <MenuItem textSecondary="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop isActive set', () => {
    const tree = renderer.create(
      <MenuItem isActive />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop openSubmenuOnMouseEnter set', () => {
    const tree = renderer.create(
      <MenuItem openSubmenuOnMouseEnter />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop subcomponentRight set', () => {
    const subcomponentRight = (
      <div> TEST </div>
        );

    const tree = renderer.create(
      <MenuItem subcomponentRight={subcomponentRight} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly without children if submenuOpen in state was set to' +
    ' true', () => {
    const component = renderer.create(
      <MenuItem text="test1" />,
    );

    const instance = component.getInstance();

    instance.setState({
      submenuOpen: true,
    });

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if call toggleMenu with param true', () => {
    const component = renderer.create(
      <MenuItem text="test1">
        <MenuGroup>
          <MenuList>
            <MenuItem text="test2" />
          </MenuList>
        </MenuGroup>
      </MenuItem>,
    );

    const instance = component.getInstance();

    instance._toggleMenu(true);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if call toggleMenu with param false', () => {
    const component = renderer.create(
      <MenuItem text="test1">
        <MenuGroup>
          <MenuList>
            <MenuItem text="test2" />
          </MenuList>
        </MenuGroup>
      </MenuItem>,
    );

    const instance = component.getInstance();

    instance._toggleMenu(false);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if call toggleMenu with not boolean param', () => {
    const component = renderer.create(
      <MenuItem text="test1">
        <MenuGroup>
          <MenuList>
            <MenuItem text="test2" />
          </MenuList>
        </MenuGroup>
      </MenuItem>,
    );

    const instance = component.getInstance();

    instance._toggleMenu(null);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('one call "handleClick"', () => {
    const mockFn = jest.fn(),
      component = renderer.create(
        <MenuItem onClick={mockFn} />,
            );

    const instance = component.getInstance();

    instance._handleClick();

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('renders correctly with children if one call "handleClick"', () => {
    const component = renderer.create(
      <MenuItem text="test1">
        <MenuGroup>
          <MenuList>
            <MenuItem text="test2" />
          </MenuList>
        </MenuGroup>
      </MenuItem>,
    );

    const instance = component.getInstance();

    instance._handleClick();

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly without children if one call "handleClick"', () => {
    const component = renderer.create(
      <MenuItem text="test1" />,
    );

    const instance = component.getInstance();

    instance._handleClick();

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly without children if "handleMouseEnter" was called' +
    ' while openSubmenuOnMouseEnter was set to true', () => {
    const component = renderer.create(
      <MenuItem openSubmenuOnMouseEnter text="test1" />,
    );

    const instance = component.getInstance();

    instance._handleMouseEnter();

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('<MenuGroup/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <MenuGroup />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop data set', () => {
    const tree = renderer.create(
      <MenuGroup data={data} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop prefix set', () => {
    const tree = renderer.create(
      <MenuGroup prefix="test" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop mode set light', () => {
    const tree = renderer.create(
      <MenuGroup mode="light" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop mode set dark', () => {
    const tree = renderer.create(
      <MenuGroup mode="dark" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop inline set true', () => {
    const tree = renderer.create(
      <MenuGroup inline />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop inline set false', () => {
    const tree = renderer.create(
      <MenuGroup inline={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop leftOffset set true', () => {
    const tree = renderer.create(
      <MenuGroup leftOffset />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop leftOffset set false', () => {
    const tree = renderer.create(
      <MenuGroup leftOffset={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});


