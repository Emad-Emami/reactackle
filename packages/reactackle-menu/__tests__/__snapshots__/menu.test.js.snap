// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> renders correctly if contains children 1`] = `
<nav
  className="MenuStyled__Menu-jtPmtm kTOSNr"
>
  <div
    className="MenuGroupStyled-koaikM ctyGEo"
  >
    <ul
      className="MenuListStyled-gFJHmJ gBqnhd"
    >
      
      <li
        className="MenuItemStyled-zHXTb cwLYjk"
        id=""
        tabIndex={-1}
      >
        <a
          active={false}
          className="AnchorStyled-cEwrfy iDSCJH"
          colorScheme="dark"
          href="#"
          nestingLevel={0}
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          
          <div
            className="ContentStyled-dbeIsI ffAfsx"
          >
            <div
              className="TextPrimaryStyled-hGxnND gdcblT"
            >
              test
            </div>
            
          </div>
          
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`<Menu/> renders correctly if prop colorScheme set dark 1`] = `
<nav
  className="MenuStyled__Menu-jtPmtm kTOSNr"
/>
`;

exports[`<Menu/> renders correctly if prop inline set true 1`] = `
<nav
  className="MenuStyled__Menu-jtPmtm kTOSNr"
/>
`;

exports[`<Menu/> renders correctly with default props 1`] = `
<nav
  className="MenuStyled__Menu-jtPmtm kTOSNr"
/>
`;

exports[`<MenuGroup/> renders correctly if prop data set 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
>
  <ul
    className="MenuListStyled-gFJHmJ gBqnhd"
  >
    
    <li
      className="MenuItemStyled-zHXTb cwLYjk"
      id=""
      tabIndex={-1}
    >
      <a
        active={false}
        className="AnchorStyled-cEwrfy iDSCJH"
        colorScheme="dark"
        href="#"
        nestingLevel={0}
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        
        <div
          className="ContentStyled-dbeIsI ffAfsx"
        >
          <div
            className="TextPrimaryStyled-hGxnND gdcblT"
          >
            Structure
          </div>
          
        </div>
        
      </a>
    </li>
    <li
      className="MenuItemStyled-zHXTb cwLYjk"
      id=""
      tabIndex={-1}
    >
      <a
        active={false}
        className="AnchorStyled-cEwrfy iDSCJH"
        colorScheme="dark"
        href="#"
        nestingLevel={0}
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        
        <div
          className="ContentStyled-dbeIsI ffAfsx"
        >
          <div
            className="TextPrimaryStyled-hGxnND gdcblT"
          >
            Design
          </div>
          
        </div>
        
        <div
          className="ExpandButtonStyled-euYyvr fjfSNm"
          onClick={[Function]}
        >
          <div
            className="ExpandIconStyled-fpoRLF dMNKzJ"
          >
            <div
              className="IconSvgStyled-cegYAl gxFYyV"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <polygon
                  points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
                />
              </svg>
            </div>
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;

exports[`<MenuGroup/> renders correctly if prop inline set false 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
/>
`;

exports[`<MenuGroup/> renders correctly if prop inline set true 1`] = `
<div
  className="MenuGroupStyled-koaikM hwNzQd"
/>
`;

exports[`<MenuGroup/> renders correctly if prop leftOffset set false 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
/>
`;

exports[`<MenuGroup/> renders correctly if prop leftOffset set true 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
/>
`;

exports[`<MenuGroup/> renders correctly if prop mode set dark 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
/>
`;

exports[`<MenuGroup/> renders correctly if prop mode set light 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
/>
`;

exports[`<MenuGroup/> renders correctly if prop prefix set 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
/>
`;

exports[`<MenuGroup/> renders correctly with default props 1`] = `
<div
  className="MenuGroupStyled-koaikM ctyGEo"
/>
`;

exports[`<MenuItem/> renders correctly if call toggleMenu with not boolean param 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test1
      </div>
      
    </div>
    
    <div
      className="ExpandButtonStyled-euYyvr fjfSNm"
      onClick={[Function]}
    >
      <div
        className="ExpandIconStyled-fpoRLF gcFwTX"
      >
        <div
          className="IconSvgStyled-cegYAl gxFYyV"
          color=""
          rotate={0}
          size="normal"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <polygon
              points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
            />
          </svg>
        </div>
      </div>
    </div>
  </a>
  <div
    className="MenuGroupStyled-koaikM ctyGEo"
  >
    <ul
      className="MenuListStyled-gFJHmJ gBqnhd"
    >
      
      <li
        className="MenuItemStyled-zHXTb cwLYjk"
        id=""
        tabIndex={-1}
      >
        <a
          active={false}
          className="AnchorStyled-cEwrfy iDSCJH"
          colorScheme="dark"
          href="#"
          nestingLevel={0}
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          
          <div
            className="ContentStyled-dbeIsI ffAfsx"
          >
            <div
              className="TextPrimaryStyled-hGxnND gdcblT"
            >
              test2
            </div>
            
          </div>
          
        </a>
      </li>
    </ul>
  </div>
</li>
`;

exports[`<MenuItem/> renders correctly if call toggleMenu with param false 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test1
      </div>
      
    </div>
    
    <div
      className="ExpandButtonStyled-euYyvr fjfSNm"
      onClick={[Function]}
    >
      <div
        className="ExpandIconStyled-fpoRLF dMNKzJ"
      >
        <div
          className="IconSvgStyled-cegYAl gxFYyV"
          color=""
          rotate={0}
          size="normal"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <polygon
              points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
            />
          </svg>
        </div>
      </div>
    </div>
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if call toggleMenu with param true 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test1
      </div>
      
    </div>
    
    <div
      className="ExpandButtonStyled-euYyvr fjfSNm"
      onClick={[Function]}
    >
      <div
        className="ExpandIconStyled-fpoRLF gcFwTX"
      >
        <div
          className="IconSvgStyled-cegYAl gxFYyV"
          color=""
          rotate={0}
          size="normal"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <polygon
              points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
            />
          </svg>
        </div>
      </div>
    </div>
  </a>
  <div
    className="MenuGroupStyled-koaikM ctyGEo"
  >
    <ul
      className="MenuListStyled-gFJHmJ gBqnhd"
    >
      
      <li
        className="MenuItemStyled-zHXTb cwLYjk"
        id=""
        tabIndex={-1}
      >
        <a
          active={false}
          className="AnchorStyled-cEwrfy iDSCJH"
          colorScheme="dark"
          href="#"
          nestingLevel={0}
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          
          <div
            className="ContentStyled-dbeIsI ffAfsx"
          >
            <div
              className="TextPrimaryStyled-hGxnND gdcblT"
            >
              test2
            </div>
            
          </div>
          
        </a>
      </li>
    </ul>
  </div>
</li>
`;

exports[`<MenuItem/> renders correctly if prop className set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop iconLeft set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="IconLeftStyled-jyAXgi faPUUg"
    >
      <div
        className="IconSvgStyled-cegYAl tgtsk"
        color=""
        rotate={0}
        size="normal"
      >
        <svg
          viewBox="0 0 24 24"
        >
          <polygon
            points="19.2,6.2 17.8,4.8 12,10.6 6.2,4.8 4.8,6.2 10.6,12 4.8,17.8 6.2,19.2 12,13.4 17.8,19.2 19.2,17.8 13.4,12"
          />
        </svg>
      </div>
    </div>
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop iconRight set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
    <div
      className="IconRightStyled-fuaQJY krUonR"
    >
      <div
        className="IconSvgStyled-cegYAl tgtsk"
        color=""
        rotate={0}
        size="normal"
      >
        <svg
          viewBox="0 0 24 24"
        >
          <polygon
            points="19.2,6.2 17.8,4.8 12,10.6 6.2,4.8 4.8,6.2 10.6,12 4.8,17.8 6.2,19.2 12,13.4 17.8,19.2 19.2,17.8 13.4,12"
          />
        </svg>
      </div>
    </div>
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop image set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="ImageStyled-kCiWXN hDCVzD"
      style={
        Object {
          "backgroundImage": "url(star)",
        }
      }
    />
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop isActive set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop linkHref set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop openSubmenuOnMouseEnter set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop prefix set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop subcomponentRight set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop text set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test
      </div>
      
    </div>
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop textRight set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    <div
      className="TextRightStyled-dhKRVv epWZfT"
    >
      test
    </div>
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly if prop textSecondary set 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly with children if one call "handleClick" 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test1
      </div>
      
    </div>
    
    <div
      className="ExpandButtonStyled-euYyvr fjfSNm"
      onClick={[Function]}
    >
      <div
        className="ExpandIconStyled-fpoRLF gcFwTX"
      >
        <div
          className="IconSvgStyled-cegYAl gxFYyV"
          color=""
          rotate={0}
          size="normal"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <polygon
              points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
            />
          </svg>
        </div>
      </div>
    </div>
  </a>
  <div
    className="MenuGroupStyled-koaikM ctyGEo"
  >
    <ul
      className="MenuListStyled-gFJHmJ gBqnhd"
    >
      
      <li
        className="MenuItemStyled-zHXTb cwLYjk"
        id=""
        tabIndex={-1}
      >
        <a
          active={false}
          className="AnchorStyled-cEwrfy iDSCJH"
          colorScheme="dark"
          href="#"
          nestingLevel={0}
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          
          <div
            className="ContentStyled-dbeIsI ffAfsx"
          >
            <div
              className="TextPrimaryStyled-hGxnND gdcblT"
            >
              test2
            </div>
            
          </div>
          
        </a>
      </li>
    </ul>
  </div>
</li>
`;

exports[`<MenuItem/> renders correctly with default props 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly without children if "handleMouseEnter" was called while openSubmenuOnMouseEnter was set to true 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test1
      </div>
      
    </div>
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly without children if one call "handleClick" 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test1
      </div>
      
    </div>
    
  </a>
</li>
`;

exports[`<MenuItem/> renders correctly without children if submenuOpen in state was set to true 1`] = `
<li
  className="MenuItemStyled-zHXTb cwLYjk"
  id=""
  tabIndex={-1}
>
  <a
    active={false}
    className="AnchorStyled-cEwrfy iDSCJH"
    colorScheme="dark"
    href="#"
    nestingLevel={0}
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
    <div
      className="ContentStyled-dbeIsI ffAfsx"
    >
      <div
        className="TextPrimaryStyled-hGxnND gdcblT"
      >
        test1
      </div>
      
    </div>
    
  </a>
</li>
`;

exports[`<MenuList/> renders correctly if prop bordered set false 1`] = `
<ul
  className="MenuListStyled-gFJHmJ gBqnhd"
>
  
</ul>
`;

exports[`<MenuList/> renders correctly if prop bordered set true 1`] = `
<ul
  className="MenuListStyled-gFJHmJ jWQriM"
>
  
</ul>
`;

exports[`<MenuList/> renders correctly if prop className set 1`] = `
<ul
  className="MenuListStyled-gFJHmJ gBqnhd"
>
  
</ul>
`;

exports[`<MenuList/> renders correctly if prop prefix set 1`] = `
<ul
  className="MenuListStyled-gFJHmJ gBqnhd"
>
  
</ul>
`;

exports[`<MenuList/> renders correctly if prop title set 1`] = `
<ul
  className="MenuListStyled-gFJHmJ gBqnhd"
>
  <div
    className="TitleStyled-kOmUSD gcEkGT"
  >
    test
  </div>
</ul>
`;

exports[`<MenuList/> renders correctly with default props 1`] = `
<ul
  className="MenuListStyled-gFJHmJ gBqnhd"
>
  
</ul>
`;
